submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/submission/domain/service/ExtSubmissionQueryService.kt

```diff
@@ -22,7 +23,12 @@ class ExtSubmissionQueryService(
     fun getReferencedFiles(accNo: String, fileListName: String): ExtFileTable {
         val sub = submissionPersistenceQueryService.getExtByAccNo(accNo, false)
         val files = submissionPersistenceQueryService.getReferencedFiles(sub, fileListName)
-        return ExtFileTable(files)
+        return ExtFileTable(files.toList())
+    }
+
+    fun getReferencedFile(accNo: String, fileListName: String): Sequence<ExtFile> {
```

Comment:
I think this should be called `getReferencedFilesAsSequece` or `getReferencedFilesSequece` since the only difference is the returned type and the current name (singular) gives the impression that this method is used to get a single file

Also, I know there're some places in the code were we use the function above (getReferencedFiles) just to get the list. I mean, we're calling the function, getting an `ExtFileTable` object, and then getting the list. Maybe it would be worth taking a look at these since it might make more sense to use this new function that returns a sequence and consume it accordingly.